bin/ember-template-lint.js

```diff
@@ -10,24 +10,58 @@ var linter = new Linter();
 const chalk = require('chalk');
 
 function printErrors(errors) {
+  const quiet = process.argv.indexOf('--quiet') !== -1;
+
   if (process.argv.indexOf('--json') + 1) {
+    let filteredErrors;
+
+    Object.keys(errors).forEach(filePath => {
+      let fileErrors = errors[filePath] || [];
+
+      filteredErrors = fileErrors.filter((error) => {
+        if (quiet && error.severity < Linter.ERROR_SEVERITY) {
+          return false;
+        }
+
+        return true;
+      });
+
+      errors[filePath] = filteredErrors;
+    });
+
     console.log(JSON.stringify(errors, null, 2));
   } else {
-    let count = 0;
+    let errorCount   = 0;
+    let warningCount = 0;
 
     Object.keys(errors).forEach(filePath => {
-      let options = {};
+      let options = {
+        quiet: quiet
+      };
       let fileErrors = errors[filePath] || [];
-      count += fileErrors.length;
+
+      errorCount   += fileErrors.filter(error => error.severity === Linter.ERROR_SEVERITY).length;
```

Comment:
Hmm, this update looks nice.  I _think_ with some small tweaks here, we can completely abstract `--quite` into the `bin/` code.

I'm thinking:

```js
let errors = fileErrors.filter(error => error.severity === Linter.ERROR_SEVERITY);
let warnings = quiet ? [] : fileErrors.filter(error => error.severity === Linter.WARNING_SEVERITY);

let errorCount = errors.length;
let warningCount = warnings.length;

// ...snip...
const messages = Linter.errorsToMessages(filePath, errors.concat(warnings), options;

// ...snip...
```

This way `Linter.errorsToMessages` _always_ is used for formatting strings from errors... 

What do you think?

Reply:
I think that's a great idea.

Reply:
Adding onto this, what are your thoughts on moving that logic above the check for the `--json` param and re-assigning errors, so we don't have to do it twice?

```js
Object.keys(errors).forEach(filePath => {
  let fileErrors = errors[filePath] || [];

  let errorsFiltered = fileErrors.filter(error => error.severity === Linter.ERROR_SEVERITY);
  let warnings = quiet ? [] : fileErrors.filter(error => error.severity === Linter.WARNING_SEVERITY);

  errorCount += errorsFiltered.length;
  warningCount += warnings.length;

  errors[filePath] = errorsFiltered.concat(warnings);
});

// check for json
// else foreach errors, console log Linter.errorsToMessages as it currently is
```

Reply:
Yep, makes sense!